src/broker/broker_config.c

```diff
@@ -1093,6 +1173,87 @@ broker_config_read_internal (const char *conf_file, T_BROKER_INFO * br_info, int
   return -1;
 }
 
+static void
+write_conf_cache (char *broker_conf_file, bool * acl_flag, int *num_broker, int *br_shm_id, char *admin_logfile,
+		  T_BROKER_INFO * br_info, time_t br_conf_mtime)
+{
+  if (broker_conf_file == NULL || num_broker == NULL || br_info == NULL || br_shm_id == NULL)
+    {
+      return;
+    }
+
+  for (int i = 0; i < MAX_NUM_BROKER_FILES; i++)
+    {
+      if (br_conf_info[i].conf_file == NULL)
+	{
+	  br_conf_info[i].conf_file = strdup(broker_conf_file);
+	  br_conf_info[i].num_broker = *num_broker;
+	  br_conf_info[i].br_shm_id = *br_shm_id;
+	  br_conf_info[i].last_modified = br_conf_mtime;
+
+	  if (acl_flag)
+	    {
+	      br_conf_info[i].acl_flag = *acl_flag;
+	    }
+
+	  if (admin_logfile)
+	    {
+	      br_conf_info[i].admin_log_file = strdup (admin_logfile);
+	    }
+
+	  memcpy (br_conf_info[i].br_info, br_info, sizeof (T_BROKER_INFO) * MAX_BROKER_NUM);
```

Comment:
If you save the broker info, isn't it right to exit the loop?

Reply:
You right! Thank you for your comment!